src/client/common/cancellation.ts

```diff
@@ -3,15 +3,16 @@
 'use strict';
 
 import { CancellationToken, CancellationTokenSource } from 'vscode';
+import { BaseError } from './errors';
 import { createDeferred } from './utils/async';
 import * as localize from './utils/localize';
 
 /**
  * Error type thrown when canceling.
  */
-export class CancellationError extends Error {
+export class CancellationError extends BaseError {
```

Comment:
Modified all of our Errors to ensure they inherit `BaseError`, 
This way we do'nt need to check each error.